### PR TITLE
5663: Add integration tests for create user validation

### PR DESCRIPTION
**Pull request by sam-roberts-bjss**

### CODE REVIEW

File: tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Features/Create User Validation.3540.feature [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+	And Users exist
+		| Id     | OrganisationName | FirstName | LastName | Email                | PhoneNumber | Disabled | Password        | OrganisationFunction |
+		| 012345 | Organisation 1   | Penny     | Lane     | PennyLane@email.com  | 01234567890 | false    | S0mePa$$w0rd    | Buyer                |
+		| 123456 | Organisation 1   | Post      | Pat      | PostmanPat@email.com | 12345678901 | false    | An0therPa$$w0rd | Authority            |
+	And a user is logged in
+		| Username             | Password        | Scope        |
+		| PostmanPat@email.com | An0therPa$$w0rd | Organisation |
+
+@3540
+Scenario: Empty first name yields a first name is required error
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Features/Create User Validation.3540.feature:20]: Are you not wanting to have numbers for each test?

Comment (sam-roberts-bjss) [reply tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Features/Create User Validation.3540.feature]: I'm not sure they provide any value. We don't do it for the unit tests. However I agree for consistency reasons I will put them back in. To be discussed in the next coding standards meeting.

File: tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Features/Create User Validation.3540.feature [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+		| Organisation 2 | Ods 2   |
+	And Users exist
+		| Id     | OrganisationName | FirstName | LastName | Email                | PhoneNumber | Disabled | Password        | OrganisationFunction |
+		| 012345 | Organisation 1   | Penny     | Lane     | PennyLane@email.com  | 01234567890 | false    | S0mePa$$w0rd    | Buyer                |
+		| 123456 | Organisation 1   | Post      | Pat      | PostmanPat@email.com | 12345678901 | false    | An0therPa$$w0rd | Authority            |
+	And a user is logged in
+		| Username             | Password        | Scope        |
+		| PostmanPat@email.com | An0therPa$$w0rd | Organisation |
+
+@3540
```
Comment (CallumBradburyBJSS) [inline tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Features/Create User Validation.3540.feature:19]: One more comment I guess, this isn't 3540 it's 3538

File: tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Features/Create User Validation.3540.feature [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+	Then a response with status code 400 is returned
+	And the response contains the following errors
+		| ErrorMessageId   | FieldName |
+		| LastNameRequired | LastName  |
+
+@3540
+Scenario: 4. A fifty one character last name yields a last name is too long error
+	When a POST request is made to create a user for organisation Organisation 2
+		| FirstName | LastName                | PhoneNumber | EmailAddress             | OrganisationName |
+		| Bob       | #a string of length 51# | 0123456789  | bob.bobkovitch@email.com | Organisation 2   |
```
[the remark this answers is not in the document]
Comment (sam-roberts-bjss) [reply tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Features/Create User Validation.3540.feature:53]: 👍

File: tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Features/Create User Validation.3540.feature (hunk 2 above)
Comment (sam-roberts-bjss) [reply tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Features/Create User Validation.3540.feature:19]: Serves me right copy & pasting 👍

**Pull request merged**